### ISSUE TITLE
transit: failed to sign payload. 403 permission denied

### ISSUE THREAD

**Issue by MRBin77** [sigstore/cosign#2646]
**Description**

When I sign a container with this Vault ACL policy `path "gitlab/*" {
   capabilities = [ "read", "list", "update" ]
}`  that's work 
![изображение](https://user-images.githubusercontent.com/26227287/214058622-0080b18b-f57f-4bc5-b466-2e1ae8ec9cd7.png)

BUT is **not secure**

When I sign a container with these Vault ACL policies
`path "gitlab/encrypt/cosign-key" {
   capabilities = [ "update" ]
}`
`path "gitlab/decrypt/cosign-key" {
   capabilities = [ "update" ]
}`
`path "gitlab/sign/cosign-key" {
   capabilities = [ "update" ]
}`
I get error 
![изображение](https://user-images.githubusercontent.com/26227287/214059826-e8d9e711-ecd5-4596-aa52-9f017de0c1b9.png)

Why does cosign need "update" permissions on the entire transit path gitlab/*? Is it a not secure.

Why doesn't cosign use a  "/sign/" policy (example `path "gitlab/sign/cosign-key" {
   capabilities = [ "update" ]
}`)?

**Version**
cosign v1.13.1/cosign-linux-amd64
cosign v1.6.0/cosign-linux-amd64

**Comment by developer-guy** [comment]
Hello @MRBin77, thanks for raising an issue! It looks like you have the same problem we discussed in Sigstore slack [here](https://sigstore.slack.com/archives/C01PZKDL4DP/p1673911582236059), if you haven't joined sigstore slack yet, [here](https://join.slack.com/t/sigstore/shared_invite/zt-1nos4nnhy-1PNlMmGLyt4eL7JB3vEJ7Q) is the join link. I'm pinging the @uralsemih, who faced the same problem before.

TLDR; you should have a variable called `VAULT_NAMESPACE.`

**Comment by MRBin77** [comment]
Following the instruction https://docs.sigstore.dev/cosign/kms_support/ I use only these variables:
VAULT_ADDR="https://vault.example.com/" 
VAULT_TOKEN=$X_VAULT_TOKEN 
TRANSIT_SECRET_ENGINE_PATH="gitlab"

I could not see anywhere in docs mentioning about VAULT_NAMESPACE

**Comment by developer-guy** [comment]
yes, the documentation will be fixed soon, can you please try this variable and confirm that whether everything is working as you expected ?

**Comment by MRBin77** [comment]
what value do i need to put in VAULT_NAMESPACE? I used VAULT_NAMESPACE="gitlab" but get same error

**Comment by developer-guy** [comment]
I'm going to hand over that question to @uralsemih

**Comment by uralsemih** [comment]
@MRBin77  As far as I understood from your question the issue is not related to `VAULT_NAMESPACE` variable. cosign is trying to use transit in the root namespace, so if you are not using any child namespace in the vault, you don't have to specify `VAULT_NAMESPACE`, if you are using a namespace config then you have to specify.

Regarding your question about `update` policy,  it is more related to Vault. There is an issue created before here as below where they require also create rather update.

https://github.com/hashicorp/vault/issues/10701

**Comment by hasanhakkaev** [comment]
MIght not be related but, as far as I understood from the screenshots , you are using GitLab CI so am I.
I have the VAULT_TOKEN in CI/CD variables like this and it worked.
<img width="456" alt="image" src="https://user-images.githubusercontent.com/14126525/214780362-a166fd80-b61a-4510-8807-dd40475e1103.png">

**Comment by MRBin77** [comment]
> MIght not be related but, as far as I understood from the screenshots , you are using GitLab CI so am I. I have the VAULT_TOKEN in CI/CD variables like this and it worked. Disabled protect and mask flags.
> 
> <img alt="image" width="456" src="https://user-images.githubusercontent.com/14126525/214780362-a166fd80-b61a-4510-8807-dd40475e1103.png">

It's not related. I have no problems with authorization.

**Comment by hasanhakkaev** [comment]
I might have a fix for this.

First of all, this is not an aswer to "why cosign needs update permissions?"  
I've a similar setup, and with a token created with the policy  below I could sign and verify with cosign.

<img width="389" alt="image" src="https://user-images.githubusercontent.com/14126525/232027600-ee08a1d5-9df2-45fa-a8a3-3872f6a500dd.png">

**Comment by MRBin77** [comment]
> I might have a fix for this.
> 
> First of all, this is not an aswer to "why cosign needs update permissions?" I've a similar setup, and with a token created with the policy below I could sign and verify with cosign.
> 
> <img alt="image" width="389" src="https://user-images.githubusercontent.com/14126525/232027600-ee08a1d5-9df2-45fa-a8a3-3872f6a500dd.png">

Delete "update" from your policy image_signing_keys/sign/* and image_signing_keys/verify/* and check could you  sign and verify with cosign? 

I need sign and verify images without "update" in policy.
